lib/features/hooks/hooks.rb

```diff
@@ -2,9 +2,51 @@
 
 require 'cucumber'
 require 'json'
+require 'securerandom'
 
 AfterConfiguration do |config|
+
+  # Start mock server
   Server.start_server
+  next if MazeRunner.configuration.farm == :none
+
+  # Setup Appium capabilities.  Note that the 'app' capability is
+  # set in a hook as it will change if uploaded to BrowserStack.
+
+  # BrowserStack specific setup
+  if MazeRunner.configuration.farm == :bs
+    tunnel_id = SecureRandom.uuid
+    MazeRunner.configuration.capabilities = Capabilities.for_browser_stack MazeRunner.configuration.device_type,
+                                                                           tunnel_id
+
+    MazeRunner.configuration.app_location = BrowserStackUtils.upload_app(MazeRunner.configuration.username,
+                                                                         MazeRunner.configuration.access_key,
+                                                                         MazeRunner.configuration.app_location)
+    BrowserStackUtils.start_local_tunnel(MazeRunner.configuration.bs_local,
+                                         tunnel_id,
+                                         MazeRunner.configuration.access_key)
+  elsif MazeRunner.configuration.farm == :local
+    MazeRunner.configuration.capabilities = Capabilities.for_local MazeRunner.configuration.device_type,
+                                                                   MazeRunner.configuration.apple_team_id,
+                                                                   MazeRunner.configuration.device_id
+  end
+  # Set app location (file or url) in capabilities
+  MazeRunner.configuration.capabilities['app'] = MazeRunner.configuration.app_location
+
+  # Create and start the drive
+  MazeRunner.driver = ResilientAppiumDriver.new(MazeRunner.configuration.appium_server_url,
+                                                MazeRunner.configuration.capabilities)
+  MazeRunner.driver.start_driver unless MazeRunner.configuration.appium_session_isolation
```

Comment:
pulling config into its own var makes this whole block more concise

```suggestion
  config = MazeRunner.configuration
  case config
  when :bs
    tunnel_id = SecureRandom.uuid
    config.capabilities = Capabilities.for_browser_stack(config.device_type, tunnel_id)

    config.app_location = BrowserStackUtils.upload_app(config.username, 
    												   config.access_key,
                                                       config.app_location)
    BrowserStackUtils.start_local_tunnel(config.bs_local, tunnel_id, config.access_key)
  when :local
    config.capabilities = Capabilities.for_local(config.device_type,
                                                 config.apple_team_id,
                                                 config.device_id)
  end
  # Set app location (file or url) in capabilities
  config.capabilities['app'] = config.app_location

  # Create and start the drive
  MazeRunner.driver = ResilientAppiumDriver.new(config.appium_server_url, config.capabilities)
  MazeRunner.driver.start_driver unless config.appium_session_isolation
```

Reply:
Yep, disappointed I didn't realise that myself.